### ISSUE TITLE
Organization and Outline

### ISSUE THREAD

**Issue by zachmay** [zachmay/inventory616#1]
### Organization

- We will keep the groups from Phase I together.
- Tasks will be assigned to individual groups, who can divide the work as they see fit.
- Groups should designate one member as their "point person". When assigning tasks, we will use GitHub's "assignee" feature, and we will assign an issue to the group's point person.
- The group consisting of me (@zachmay), Paul, and Steve will work primarily to divide up work, coordinate effort, and review and test changes. Since we are going to work from our Phase I proposal, feel free to ask us questions about that.
- Paul will probably coordinate work on the API.
- Steve will probably coordinate work on the front-end.
- Zach is *not* mad with power.
- (Yes he is.)

### Dev Environment

- Who has tried setting up? Issues?
- If you have set it up, destroy the VM and recreate it. :)

### Assignments

- [Phase I Document](https://docs.google.com/a/sector42.net/document/d/17LdthtmzmdrGW1asYWCBkHTMn2I0r00SqIjjt1Tk2cE/edit#); this will be the reference for what we're building.
- PHP experience? Who will concentrate on the API?
- JavaScript experience? Preference for Angular or Backbone?
- Automated testing? Volunteers for a team of "test engineers"?

### How to get stuff done

[See this wiki page](https://github.com/zachmay/inventory616/wiki/Development-Procedure)

### TODO

Everyone needs to post a comment to this issue. Identify yourself as point person for your group if that's you.

**Comment by xuehan6678** [comment]
Thanks for the setup work.
I'm interested in doing the automated testing part.

**Comment by pjlinton** [comment]
I'm here from team #2 - @pjlinton

**Comment by pjlinton** [comment]
Group #2 Paul Linton

**Comment by ForeverTangent** [comment]
Feet and Fish

**Comment by sjcurt02** [comment]
Steven Curtis
Group number 2
User interface implementation
GitHub user ID: sjcurt02

**Comment by ybwirian** [comment]
I'm in -Yosef (Group 1)-

**Comment by zachmay** [comment]
Group 1 - Front-end
Group 3 - Front-end
Group 4 (Peng) - PHP API
Group 5 - API test suite
Group 6 - PHP API

16 checkins + Group 2

**Comment by tttr222** [comment]
Tung Tran 
Team 6

**Comment by hepengcs** [comment]
Peng
Team4

**Comment by wwen0** [comment]
I'm in. 
Wei Wen (Team 4)

**Comment by lixinchengdu** [comment]
Xin Li
Team 5

**Comment by liudw9094** [comment]
Dawei Liu
Team 5

**Comment by nedjem** [comment]
Shruti Sinha
Team 3

**Comment by jdhuns2** [comment]
James Hunsucker
Team 6

**Comment by nastiakazadi** [comment]
Anastasia Kazadi
Team 1

**Comment by akm-sabbir** [comment]
Team 6, Sabbir

**Comment by zachmay** [comment]
@nedjem Can you get the rest of your team members to check in here? I talked Anirban earlier today about setting up the VM, but I don't have his GitHub username.

**Comment by zachmay** [comment]
@liudw9094 @lixinchengdu, I think I'm missing one more person from your group. Can you have him or her check in? Thanks!

**Comment by huazhewang** [comment]
Sorry for late. 
Huazhe Wang
Team 5

**Comment by zachmay** [comment]
@huazhewang No problem. :)

**Comment by harika05** [comment]
A.L.P Harika Team 3. Sorry for the delay.

**Comment by harika05** [comment]
@zachmay Yeah Sure, will ask them to do so ASAP.

**Comment by nedjem** [comment]
@zachmay Sure will tell them to do so ASAP.

**Comment by zachmay** [comment]
@harika05 No worries. Thanks, @nedjem! :+1:

**Comment by zhangyimc** [comment]
SO sorry for late
This is Yi Zhang from Team 5
Thanks

**Comment by zachmay** [comment]
Good morning, everyone. I'm going to post assignments later today.

**Comment by zachmay** [comment]
Hello! I've posted a series of tasks that we'll be working on in the 'Issues' section of the repo. I've assigned teams using the issue "labels". You should probably "check-in" over on the appropriate issues so that you'll be notified whenever someone posts a message to the issue assigned to your team.

Important to note: not all of these issues are of the same size. In particular, the API issues are broken into smaller, "cohesive" chunks. When a group gets done with one of those, I'll assign them to another. The ones assigned now are the highest priority ones, I think.

Prior to next class, look at your assignments, or even try your hand at digging into the code.

**Comment by anirban15** [comment]
Thanks Zach, Anirban (Group 3)

**Comment by zachmay** [comment]
Hey everyone. If you haven't set up your dev environment yet, I added one minor point to the setup instructions: you'll need to set up your SSH keys for GitHub. It's pretty easy.

Thanks to @akm-sabbir for bringing this up!

**Comment by zachmay** [comment]
I promised the API teams a link to a really good resource on the Laravel framework we're using: [Laracasts](https://laracasts.com/series/laravel-5-fundamentals).

The first video has some good background on The PHP ecosystem and the second can safely be skipped. The good stuff starts in video 3!

**Comment by nedjem** [comment]
@zachmay  Thanks!

**Comment by wwen0** [comment]
Thanks, Zach. This is very helpful

**Comment by zachmay** [comment]
@wwen0 Good to hear!

Also, I think you guys can safely skip the material about templates (Blade 101, etc.), since the API's output is just JSON and Laravel knows how to serialize a PHP associative array into JSON automatically.

**Comment by zachmay** [comment]
I'm going to be meeting with some folks in the common area of the Ralph G. Anderson building at 5:30pm today (might be there a bit earlier than that). This is primarily for the PHP/API teams, but if you're in the area and have questions, come on by.

**Comment by wwen0** [comment]
@zachmay  Very good advice. Otherwise, I will waste a lot more time. BTW, who will be responsible for creating and populating the database?

**Comment by hepengcs** [comment]
I'd like to introduce a great tool for RESTful API testing called Postman. It can help you to send various RESTful requests with Json and XML data. Here is a introduction video.
https://www.youtube.com/watch?v=JZP2ose-OBQ

**Comment by zachmay** [comment]
Hey everyone. We've made some comments on the pull requests from teams 1, 4, and 6. We're still waiting for a pull request from team 3 and test plan documentation from team 5.

The next time you start working on the project, be sure to pull down the latest changes. `git fetch origin` will grab everything, then merge `master` into your branch. This will pick up a change I made to disable Laravel's CSRF protection by default (which has caused some of the API folks issues).

Thanks everyone!
